client/question/services/FeedbackGeneratorService.js

```diff
@@ -417,21 +515,56 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // If the user receives the same error message increment the same error
+        // counter.
+        if (previousErrorString &&
+            previousErrorString === codeEvalResult.getErrorString()) {
+          _updateCounters(ERROR_COUNTER_SAME_RUNTIME);
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        _applyThresholdUpdates(feedback);
+        previousErrorString = codeEvalResult.getErrorString();
         return feedback;
       },
+      /**
+       * Returns the Feedback object for the given syntax error string.
+       *
+       * @param {string} errorString
+       * @returns {Feedback}
+       */
       getSyntaxErrorFeedback: function(errorString) {
+        // If the user receives another syntax error, increment the
+        // language unfamiliarity error counter.
+        _updateCounters(ERROR_COUNTER_LANGUAGE_UNFAMILIARITY);
+        previousErrorString = '';
```

Comment:
Is this "error string" supposed to be for general errors or just the specific case of runtime errors? If the latter, perhaps it should be renamed to previousRuntimeErrorString.